miner/auto_agent.go

```diff
@@ -0,0 +1,142 @@
+// Copyright 2015 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package miner
+
+import (
+	"sync"
+	"time"
+
+	"sync/atomic"
+
+	"github.com/ethereumproject/go-ethereum/common"
+	"github.com/ethereumproject/go-ethereum/core/types"
+	"github.com/ethereumproject/go-ethereum/logger"
+	"github.com/ethereumproject/go-ethereum/logger/glog"
+	"github.com/ethereumproject/go-ethereum/pow"
+)
+
+type AutoAgent struct {
+	mu sync.Mutex
+
+	workCh        chan *Work
+	quit          chan struct{}
+	quitCurrentOp chan struct{}
+	returnCh      chan<- *Result
+
+	index int
+	pow   pow.PoW
+
+	isMining int32 // isMining indicates whether the agent is currently mining
+}
+
+func NewAutoAgent(index int) *AutoAgent {
+	miner := &AutoAgent{
+		index: index,
+	}
+
+	return miner
+}
+
+func (self *AutoAgent) Work() chan<- *Work            { return self.workCh }
+func (self *AutoAgent) Pow() pow.PoW                  { return self.pow }
+func (self *AutoAgent) SetReturnCh(ch chan<- *Result) { self.returnCh = ch }
+
+func (self *AutoAgent) Stop() {
+	self.mu.Lock()
+	defer self.mu.Unlock()
+
+	close(self.quit)
+}
+
+func (self *AutoAgent) Start() {
```

Comment:
`self` is considered bad name in go. `aa` or something more meaningful is preferred (I need a reference for this ;))

Reply:
Yea -- will fix for all the *agents.go. I just copy and pasted the remote_agent file which had that.